source/common/router/config_impl.cc

```diff
@@ -855,5 +878,27 @@ ConfigImpl::ConfigImpl(const envoy::api::v2::RouteConfiguration& config, Runtime
                                                      config.response_headers_to_remove());
 }
 
+PerFilterConfigs::PerFilterConfigs(const Protobuf::Map<std::string, ProtobufWkt::Struct>& configs) {
+  for (const auto& cfg : configs) {
+    const std::string& name = cfg.first;
+    const ProtobufWkt::Struct& struct_config = cfg.second;
+
+    auto& factory = Envoy::Config::Utility::getAndCheckFactory<
+        Server::Configuration::NamedHttpFilterConfigFactory>(name);
+
+    configs_[name] = Envoy::Config::Utility::translateToFactoryRouteConfig(struct_config, factory);
+  }
+}
+
+const Protobuf::Message* PerFilterConfigs::get(const std::string& name) const {
+  auto cfg = configs_.find(name);
+
+  if (cfg == configs_.end()) {
```

Comment:
nit: I would just use the ternary operator here to collapse the rest of these lines into one.